test/flow/codeGeneration.js

```diff
@@ -0,0 +1,312 @@
+
+
+        export interface HeroNameQueryVariables {
+          episode: ? Episode;
+        }
+
+        export interface HeroNameQuery {
+          hero: ? {
+            name: string,
+          };
+        }
+      `);
+    });
+
+    it(`should generate simple nested query operations including input variables`, function() {
+      const context = this.compileFromSource(`
+        query HeroAndFriendsNames($episode: Episode) {
+          hero(episode: $episode) {
+            name
+            friends {
+              name
+            }
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        //  This file was automatically generated and should not be edited.
+
+        // The episodes in the Star Wars trilogy
+        export type Episode =
+          "NEWHOPE" | // Star Wars Episode IV: A New Hope, released in 1977.
+          "EMPIRE" | // Star Wars Episode V: The Empire Strikes Back, released in 1980.
+          "JEDI"; // Star Wars Episode VI: Return of the Jedi, released in 1983.
+
+
+        export interface HeroAndFriendsNamesQueryVariables {
+          episode: ? Episode;
+        }
+
+        export interface HeroAndFriendsNamesQuery {
+          hero: ? {
+            name: string,
+            friends: ? Array< {
+              name: string,
+            } >,
+          };
+        }
+      `);
+    });
+
+    it(`should generate fragmented query operations`, function() {
+      const context = this.compileFromSource(`
+        query HeroAndFriendsNames {
+          hero {
+            name
+            ...HeroFriends
+          }
+        }
+
+        fragment HeroFriends on Character {
+          friends {
+            name
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        //  This file was automatically generated and should not be edited.
+
+        export interface HeroAndFriendsNamesQuery {
+          hero: ? HeroFriendsFragment & {
+            name: string,
+          };
+        }
+
+        export interface HeroFriendsFragment {
+          friends: ? Array< {
+            name: string,
+          } >;
+        }
+      `);
+    });
+
+    it(`should generate query operations with inline fragments`, function() {
+      const context = this.compileFromSource(`
+        query HeroAndDetails {
+          hero {
+            name
+            ...HeroDetails
+          }
+        }
+
+        fragment HeroDetails on Character {
+          ... on Droid {
+            primaryFunction
+          }
+          ... on Human {
+            height
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        //  This file was automatically generated and should not be edited.
+
+        export interface HeroAndDetailsQuery {
+          hero: ? HeroDetailsFragment & {
```

Comment:
I've run into issues before with Flow's intersection types, particularly when there are overlapping fields (which is totally valid between fragments in GraphQL). Is there any way to add actual flow tests here that the generated code type checks?

Otherwise I imagine that these tests are going to be maintained by simply copying the generated output as it evolves, but ideally the crux of the testing would be that the generated types are correctly constrained.